src/App/Pages/Vault/ScanPageViewModel.cs

```diff
@@ -39,6 +53,23 @@ public bool ShowScanner
                 });
         }
 
+        private async Task ToggleScanMode()
+        {
+            var cameraPermission = await PermissionManager.CheckAndRequestPermissionAsync(new Permissions.Camera());
+            HasCameraPermission = cameraPermission == PermissionStatus.Granted;
+            if (!HasCameraPermission)
+            {
+                var openAppSettingsResult = await _platformUtilsService.ShowDialogAsync(AppResources.EnableCamerPermissionsToUseTheScanner, title: string.Empty, confirmText: AppResources.Settings, cancelText: AppResources.NoThanks);
+                if (openAppSettingsResult)
+                {
+                    _deviceActionService.OpenAppSettings();
+                    return;
+                }
+            }
+            ShowScanner = !ShowScanner;
+            InitScannerCommand.Execute(null);
```

Comment:
I don't like much having a command that has its action on the view; I think it might cause to never dispose correctly the ViewModel because of the strong reference to its action.
We could have something like [MvxInteraction](https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross/ViewModels/MvxInteraction.cs) of MvvmCross but we'll need to discuss it and address this throughout the app. For now we can leave it like that given that there are already other things that hold strong references